domain/application/resource/store.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package resource
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/juju/juju/core/objectstore"
+	"github.com/juju/juju/internal/charm/resource"
+)
+
+// ResourceStore provides a list of methods necessary for interacting with
+// a store for the resource.
+type ResourceStore interface {
+	// Get returns an io.ReadCloser for data at path, namespaced to the
```

Comment:
This comment should probably be resource specific, i.e. `Get returns the size and an IO reader for the specified resource from the resource store`

Reply:
Let's adjust the comment in your implementation PR, this is the stub code.